modin/pandas/concat.py

```diff
@@ -108,8 +110,15 @@ def concat(
             new_idx_labels = {
                 k: v.index if axis == 0 else v.columns for k, v in zip(keys, objs)
             }
-            tuples = [(k, o) for k, obj in new_idx_labels.items() for o in obj]
+            tuples = [
+                (k, *o) if isinstance(o, tuple) else (k, o)
```

Comment:
In what cases `o` will be a `tuple`.

Reply:
Roughly say here we iterating through indices of `objs`, so if `obj` has MultiIndex index, then its values will be a `tuple`